### DISCUSSION TITLE
Question about cancellation CQE ordering

### DISCUSSION THREAD

**Discussion by bnbajwa** [axboe/liburing#900] [Q&A] [answered]
Suppose I submit a SQE and lets call it A.
And suppose that I submit a SQE to cancel the previous SQE, lets call this SQE B.

If cancellation succeeds, then we will get 2 CQEs:
1) CQE for SQE A with error ECANCELED
2) CQE for SQE B with result success

I was wondering if there is any ordering for the 2 CQEs? Or can the 2 CQEs arrive in any order?

Thanks

**Comment by axboe** [comment, marked as answer]
Cancelation can also succeed async, eg if the request is queued with io-wq. Normally it would indeed have the order you listed, but this is not a guarantee and an application should not rely on it.
